### ISSUE TITLE
repetitive entries and conflicting answers

### ISSUE THREAD

**Issue by XinhaoMei** [soham97/mellow#3]
Hi, congratulations on this amazing work! I tried to use the dataset, but found that it contains some repetitive entries and conflicting answers, particularly in the ClothoAQA subset. Here's an example from the train split:

```
{'taskname': 'clotho_aqa_train', 'filepath1': 'ClothoAQA\\audio_files\\AmbianceBackyard_Quiet_bip.wav', 'filepath2': '', 'caption1': '', 'caption2': '', 'input': 'Are these animals outside? Answer only using yes or no.', 'answer': 'no', 'subtype': 'ClothoAQA.json'}
{'taskname': 'clotho_aqa_train', 'filepath1': 'ClothoAQA\\audio_files\\AmbianceBackyard_Quiet_bip.wav', 'filepath2': '', 'caption1': '', 'caption2': '', 'input': 'Are these animals outside? Answer only using yes or no.', 'answer': 'yes', 'subtype': 'ClothoAQA.json'}
{'taskname': 'clotho_aqa_train', 'filepath1': 'ClothoAQA\\audio_files\\AmbianceBackyard_Quiet_bip.wav', 'filepath2': '', 'caption1': '', 'caption2': '', 'input': 'Are these animals outside? Answer only using yes or no.', 'answer': 'yes', 'subtype': 'ClothoAQA.json'}
```

Could you please clarify how these should be handled?

Thank you very much!

**Comment by soham97** [comment]
Hi @XinhaoMei, thank you for checking the project! The subtype entry for `ClothoAQA.json` uses audio files and labels from the ClothoAQA dataset [1]. Each question in ClothoAQA is annotated by three individuals, whose responses occasionally differ. The CLothoAQA authors explored several approaches: directly using the raw data, applying majority voting, or retaining only data points with unanimous agreement among the annotators (see Table 1 in the paper).

For Mellow, we used the unfiltered data as-is, aiming for the model to learn to disregard noise during pretraining. However, for each ClothoAQA entry, using majority voting or filtering for unanimous responses might get you better results. Unfortunately, we did not experiment on this to provide any concrete answer on what works better.

[1] [https://arxiv.org/abs/2204.09634](https://arxiv.org/abs/2204.09634)

**Comment by XinhaoMei** [comment]
Got it. Thanks for the clarification!
